pkg/hpa/service.go

```diff
@@ -765,3 +766,54 @@ func (c *Service) excludeExternalMetric(ctx context.Context, hpa *v2.HorizontalP
 
 	return newHPA
 }
+
+func (c *Service) CheckHpaMetricStatus(ctx context.Context, currenthpa *v2.HorizontalPodAutoscaler) bool {
+	//currenthpa = currenthpa.DeepCopy()
+	logger := log.FromContext(ctx)
+	if currenthpa == nil {
+		logger.Info("empty HPA passed into status check, ignore")
+		return true
+	}
+
+	if reflect.DeepEqual(currenthpa.Status, v2.HorizontalPodAutoscalerStatus{}) {
+		logger.Info("HPA empty status, switch to emergency mode")
+		return false
+	}
+
+	if currenthpa.Status.Conditions == nil {
+		logger.Info("HPA empty conditions, switch to emergency mode")
+		return false
+	}
+
+	if currenthpa.Status.CurrentMetrics == nil {
+		logger.Info("HPA no metrics, switch to emergency mode")
+		return false
+	}
```

Comment:
and, when `err`, we should abort the reconciliation because HPA's object looks like not valid at all.

Reply:
Right, looks like it's expected that if HPA is just created, Status etc could be empty (since HPA controller might not yet notice a new HPA)
Can we early-return (or not run this function) when tortoise's phase is `Initializing`?

Reply:
yes i handled it to only run when tortoise is working. But there is a test case where tortoise is working but no hpa and tortoise reconcile should create a hpa. This will always fail and im not sure of a workaround for this specific test case

Reply:
Can we just add a HPA then? Tortoise with `Working` phase, but doesn't have HPA looks not a correct scenario.

Reply:
i think it should switch the phase when a new hpa is created? maybe to partlyworking or initializing?

Reply:
No, if HPA is created at `UpdateHPASpecFromTortoiseAutoscalingPolicy`

Reply:
Rethinking about `UpdateHPASpecFromTortoiseAutoscalingPolicy`, the tortoise controller could add a new metric based on the update in tortoise's spec, which could result in missing metrics but we should turn the emergency mode on.

Reply:
yes but because of that it will cause error since it generates hpa with empty status and tortoisephase is still working. A rather hackish way around it is if a new hpa is created from that function it returns a bool so we skip hpa status check on that reconcile loop and check it on future reconcile loops

Reply:
Hmmm, that's complicated.
Okay, I now agree that we don't return error from this function even when HPA has empty condition etc, and we just ignore, doing nothing at those cases.

Reply:
regarding the missing metrics concern as well, would skipping the check if theres a change in policy be a good workaround? as long as the metrics are populated by the next reconcile loop it should work?

Reply:
There's two cases coming from `UpdateHPASpecFromTortoiseAutoscalingPolicy`, IIUC:
1. A new HPA is created: In this case, we skip. So, no issue.
2. A new metric is added: In this case, a status doesn't have anything about new metric added at HPA spec, but this function doesn't check missing metrics by comparing spec and status, so no issue.

So, I don't see any issues. WDYT

Reply:
i dont see issues as well just that it may not be the cleanest way to code this? But I can't think of anything better at the moment